### PR TITLE
Closes #4911: Adds downloadStatus to onDownloadCompleted

### LINKED ISSUE #4911: Add status of download in `onDownloadCompleted`
For Fenix issue: https://github.com/mozilla-mobile/fenix/issues/5958 we will need to know whether the download failed or completed successfully when we get the `onDownloadCompleted` callback. Because of this, we should pass that status along to the consumer.

### PR DESCRIPTION
**Pull request by sblatz**
---

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt [hunk 1]
```diff
@@ -170,35 +174,37 @@ abstract class AbstractFetchDownloadService : Service() {
...
+                    currentDownloadJobState.status = DownloadJobStatus.COMPLETED
                     DownloadNotification.createDownloadCompletedNotification(context, download)
                 }
 
                 DownloadJobStatus.FAILED -> {
                     DownloadNotification.createDownloadFailedNotification(context, download)
                 }
 
-                null -> { return }
+                else -> { return }
```
Comment (NotWoods) [inline components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt]: nit: no braces
```suggestion
                else -> return
```

File: components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt [hunk 2]
```diff
@@ -352,6 +359,7 @@ abstract class AbstractFetchDownloadService : Service() {
         private const val PARTIAL_CONTENT_STATUS = 206
         private const val OK_STATUS = 200
 
+        const val EXTRA_DOWNLOAD_STATUS = "mozilla.components.feature.downloads.EXTRA_DOWNLOAD_STATUS"
```
Comment (NotWoods) [inline components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt]: nit:
```suggestion
        const val EXTRA_DOWNLOAD_STATUS = "mozilla.components.feature.downloads.DOWNLOAD_STATUS"
```
or
```suggestion
        const val EXTRA_DOWNLOAD_STATUS = "mozilla.components.feature.downloads.extras.DOWNLOAD_STATUS"
```

**Comment by sblatz** [comment]
bors r=@NotWoods

**Comment by sblatz** [comment]
bors retry

**Comment by sblatz** [comment]
Bors r=@NotWoods

**Comment by sblatz** [comment]
bors retry

**Comment by sblatz** [comment]
@JohanLorenzo This PR is stuck in Bors hell like we used to see on Fenix, any ideas what's going on?

**Comment by grigoryk** [comment]
This is hitting multiple intermittent test failures :(

**Comment by grigoryk** [comment]
Intermittents this hit so far: #4929 https://github.com/mozilla-mobile/android-components/issues/4928

**Comment by sblatz** [comment]
bors r=@NotWoods

**Pull request merged**